packages/task/src/browser/task-service.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { ILogger } from '@theia/core/lib/common';
+import { FrontendApplication } from '@theia/core/lib/browser';
+import { ITaskServer, ITaskExitedEvent, ITaskOptions, ITaskInfo } from '../common/task-protocol';
+import { TERMINAL_WIDGET_FACTORY_ID, TerminalWidgetFactoryOptions } from '@theia/terminal/lib/browser/terminal-widget';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import { TaskWatcher } from '../common/task-watcher';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+import { TaskConfigurations, TaskConfigurationClient } from './task-configurations';
+import { TerminalWidget } from '@theia/terminal/lib/browser/terminal-widget';
+
+@injectable()
+export class TaskService implements TaskConfigurationClient {
+
+    protected workspaceRootUri: string | undefined = undefined;
+
+    constructor(
+        @inject(FrontendApplication) protected readonly app: FrontendApplication,
+        @inject(ITaskServer) protected readonly taskServer: ITaskServer,
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(TaskWatcher) protected readonly taskWatcher: TaskWatcher,
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
+        @inject(TaskConfigurations) protected readonly taskConfigurations: TaskConfigurations
+    ) {
+        // wait for the workspace root to be set
+        this.workspaceService.root.then(root => {
+            if (root) {
+                this.taskConfigurations.watchWorkspaceConfiguration(root.uri);
+                this.workspaceRootUri = root.uri;
+            }
+        });
+
+        // notify user that task has started
+        this.taskWatcher.onTaskCreated((event: ITaskInfo) => {
+            // is event interesting to this client?
+            if (event.ctx === this.getContext()) {
+                this.messageService.info(`Task #${event.taskId} created - ${event.label}`);
+            }
+        });
+
+        // notify user that task has finished
+        this.taskWatcher.onTaskExit((event: ITaskExitedEvent) => {
+            const signal = event.signal;
+            // is event interesting to this client?
+            if (event.ctx === this.getContext()) {
+                if (event.code === 0) { /* normal process exit */
+                    let success = '';
+
+                    if (signal) {
```

Comment:
Yes, there is room for improvement here. On Windows, interrupting a task using Ctrl-C results in a non-zero exit code and undefined signal. On Linux, the exit code is success, with a specific signal. It would be nice if we could abstract those differences.